### DISCUSSION TITLE
TA on Raspberry and Roland V-1200HD

### DISCUSSION THREAD

**Discussion by feta00** [josephdadams/TallyArbiter#657] [Q&A] [answered]
Hi everyone and thanks for developing TA.
I had success in using it installed in a windows based environment with a Roland V-160HD with Smart Tally and a bunch of ATOMs..

I was wondering if I could implement it on a different environment but I need your help out there! 
Firstly, I'd like to run TA on a raspberry, is it possible to use the appimage installer on this ARM computers? I'm not a pro with the terminal..

Secondly, this other environment implies a Roland V-1200HD video mixer, which is not provided with Smart Tally functionality, but uses a D-Sub 25 pin TALLY/GPIO Connector with this specifications: 

> Tally/control output connector specifications
> Maximum input: 12 V/200 mA
> Open-collector type

Do I need a relay board to be connected through USB to raspberry? I'm not sure what they mean with open-collector but I suspect it does not mean contact-closures.. 
Or must I connect the D_Sub directly to the GPIO connectors on the Raspberry?!

Sorry for such basics questions but I'm kind of a beginner in this specific fields.
Thanks for any help.

**Comment by chibiconsulting** [comment]
FYI: Open collector means a connection to ground, typically via a transistor, optical isolator or relay. The 12V/200mA is the most voltage/current it can handle.
Since Raspberry input operates at 3.3v with minimal current, this is much less voltage/current than the capacity, so it could be interfaced to inputs on the Raspberry without issue; however, if you are not comfortable with wiring it up yourself, an external device might be easier for you. Either way can be made to work.
I do not have current experience with specific devices on the market; however, there are many people who do, so you should get more responses soon.

**Reply by feta00** [reply]
Hi, I was reading the relay board documentation and there is something I'd like to be sure and you seem the right one to clarify things up!
Sorry if it might be a dumb question but as I said I'm a beginner.

When you say "connection to ground" you mean I need to wire each pin (1-16 in the picture below) from the D-Sub connector to the relay's COM input? So I would not need the other inputs (NO and NC)? 
But then what's for PIN17 (GND)?

Thanks for any help.

![Screenshot 2024-02-29 161556](https://github.com/josephdadams/TallyArbiter/assets/107259076/ea4a24ad-e6dc-4fcf-a58e-7fc280b12935)
![Screenshot 2024-02-29 163131](https://github.com/josephdadams/TallyArbiter/assets/107259076/9e6391fa-ff44-4539-b423-9f676e4556f8)

**Reply by josephdadams** [reply]
Contact closure and GPIO are not the same thing. A relay (contact closure) is like taking two wires and temporarily touching them together or disconnecting them.

In your case you need to actually send a small amount of voltage down the line, so you will need to use something like a raspberry Pi with GPIO pins capable of this.

**Reply by feta00** [reply]
Ok, thanks for clarification! I'm really unexperienced in this! Apologies!

Problem is my skills with CLI and linux are very basic so I would probably have a hard time setting everything up even with the script you were mentioning you could license me. 

I doubt it but it's probably worth asking before giving up! 
Would a GPIO USB board work on a windows system with TA installed on?

Thanks for your time and for developing TA!

**Reply by josephdadams** [reply]
A GPIO usb board would not work out of the box since there isn’t code for that currently.

**Reply by feta00** [reply]
As I thought, thanks for replying. Appreciated.

**Comment by josephdadams** [comment]
Yes you can use the GPIO pins on a Pi.

Follow the CLI instructions in the documentation to install via the terminal.

TA currently has no native support for GPIO input data.

I have a GPIO to TSL 3.1 script that I could license you. Email me directly to discuss that. (My email is listed on my GitHub profile.)

**Comment by feta00** [comment, marked as answer]
Thanks for replying..
I will try using a relay board that it seems the easyest option.
I'll let you know.
